### DISCUSSION TITLE
Change ssl certificate without dropping connections

### DISCUSSION THREAD

**Discussion by abhay-s-rawat** [uNetworking/uWebSockets.js#895] [Q&A]
Hi , I am using uWebSockets.js with socketio and I was searching for way to update ssl certificate when ssl renews without restarting server. Like its done with httpserver. I can use filewatcher to know if certificate is renewed, can you please tell how to change in uWebSockets.js.

```
httpServer.setSecureContext(
{
  key: fs.readFileSync(settings.PRIVKEY),
  cert: fs.readFileSync(settings.FULLCHAIN)
});
```
```
const { SSLApp } = require("uWebSockets.js");
const { Server } = require("socket.io");

const app = new SSLApp({
  key_file_name: "keys/server_key.pem",
  cert_file_name: "keys/server_cert.pem",
});
const io = new Server({ transports: ["websocket"] });
io.attachApp(app);
```

**Comment by uNetworkingAB** [comment]
You can use the SNI feature for this.

**Reply by snowinszu** [reply]
> You can use the SNI feature for this.

What is SNI feature?

**Reply by denyaalt** [reply]
> You can use the SNI feature for this.

How to use SNI feature if addServerName() just adds new certificate, there is no function to update?

**Comment by slvsfn** [comment]
I use uWebsocket.js, I have the same problem, let's encrypt with Cerbot automatically updates the server certificate, SSLApp keeps the data of the expired certificate.
